tools/ports/__init__.py

```diff
@@ -426,16 +428,53 @@ def show_port_help_and_exit(port):
   sys.exit(0)
 
 
-def handle_use_port_arg(settings, arg, error_handler=None):
-  if not error_handler:
-    def error_handler(message):
-      handle_use_port_error(arg, message)
+# extract dict and delegate to port.handle_options for handling (format is 'option1=value1:option2=value2')
+def handle_port_options(name, options, error_handler):
+  port = ports_by_name[name]
+  if options == 'help':
+    show_port_help_and_exit(port)
+  if not hasattr(port, 'handle_options'):
+    error_handler(f'no options available for port `{name}`')
+  else:
+    options_dict = {}
+    for name_value in options.split(':'):
+      nv = name_value.split('=', 1)
+      if len(nv) != 2:
+        error_handler(f'`{name_value}` is missing a value')
+      if nv[0] not in port.OPTIONS:
+        error_handler(f'`{nv[0]}` is not supported; available options are {port.OPTIONS}')
+      if nv[0] in options_dict:
+        error_handler(f'duplicate option `{nv[0]}`')
+      options_dict[nv[0]] = nv[1]
+    port.handle_options(options_dict, error_handler)
+
+
+# handle port dependencies (ex: deps=['sdl2_image:formats=jpg'])
+def handle_port_deps(name, error_handler):
+  port = ports_by_name[name]
+  for dep in port.deps:
+    dep_name, dep_options = split_port_options(dep)
+    if dep_name not in ports_by_name:
+      error_handler(f'unknown dependency `{dep_name}`')
+    if dep_options:
+      handle_port_options(dep_name, dep_options, error_handler)
+    handle_port_deps(dep_name, error_handler)
```

Comment:
Rather than recursively traversing the deps here, and duplicating the `unknown dependency` error from `resolve_dependencies`, can we just use `resolve_dependencies` here and then iterate the list without recursion?

Now that I mention this it looks like maybe we don't handle port options for except for `handle_use_port_arg`... so if we just do `-sUSE_SDL` or something like that then we any options in the transitive dependencies would be ignored.  I guess this is fine as long as only contrib ports have deps with options?

Reply:
I think I am definitely leaning towards supporting this only for the `--use-port` syntax. At this moment, none of the "old" ports are relying on a new port that requires the `--use-port` syntax, so it is not like this use case even exists.

So I think the current implementation is fine. Due to the call `node.process_dependencies(settings)` in `resolve_dependencies` which as I pointed in my previous PR has side effects, I would rather not call it in this implementation.